grails-app/controllers/au/org/ala/volunteer/TutorialsController.groovy

```diff
@@ -352,21 +358,33 @@ class TutorialsController {
         }
 
         def institutionFilter = []
-        Institution institution = (params.institutionFilter ? Institution.get(params.long('institutionFilter')) : null)
+        def institutionId = params.long('institutionFilter') ?: getSessionFilter("institutionFilter")
+        Institution institution = institutionId ? Institution.get(institutionId as long) : null
+
         if (institution) {
             institutionFilter.add(institution)
-            updateSessionFilter([institutionFilter: params.institutionFilter])
+            params.institutionFilter = institutionId
+            updateSessionFilter([institutionFilter: institutionId])
         } else {
             institutionFilter = institutionList
             removeSessionFilter("institutionFilter")
         }
 
-        def statusFilter = (params.statusFilter ?: null)
-        if (params.statusFilter) updateSessionFilter([statusFilter: params.statusFilter])
-        else removeSessionFilter("statusFilter")
+        def statusFilter = params.statusFilter ?: getSessionFilter("statusFilter")
+        if (statusFilter) {
+            params.statusFilter = statusFilter
+            updateSessionFilter([statusFilter: statusFilter])
+        } else {
+            removeSessionFilter("statusFilter")
+        }
 
-        if (params.q) updateSessionFilter([q: params.q])
-        else removeSessionFilter("q")
+        if (params.q) {
+            updateSessionFilter([q: params.q])
+        } else if (getSessionFilter("q")) {
+            params.q = getSessionFilter("q")
+        } else {
+            removeSessionFilter("q")
```

Comment:
As above I don't think this is going to be reached.  As as aside this expression of the behaviour was easier to understand than the above version, I'd consider refactoring this into a method that makes sure the param and session value are in sync and having a couple of calls to it here.

Reply:
See previous reply.